memmetrics/roundtrip.go

```diff
@@ -138,6 +148,10 @@ func (m *RTMetrics) Append(other *RTMetrics) error {
 		return err
 	}
 
+	m.statusCodesLock.Lock()
```

Comment:
I think there's a potential deadlock here:

a.Append(b)

and 

b.Append(a)

will deadlock.

To mitigate the issue, I would suggest to do the following:

1. implement Export function that uses RLock and returns the copy of the map with status codes
2. in append, do this:

```go

copied := other.Export()
m.Lock()
defer m.Unlock()
do the merge....
return 
```